### PR TITLE
feat: adds header slots

### PR DESCRIPTION
**Pull request by gusnaughton**
# Alaska Airlines Pull Request

Adds a header slot to auro-flight for displaying information such as Scheduled departures

image

## Type of change:

Please delete options that are not relevant.

- [x] New capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by gusnaughton** [comment]
image

File: demo/demo.md [hunk 1]
```diff
@@ -68,6 +68,46 @@ This example illustrates a mainline nonstop flight from SEA to EWR (SEAEWR)
 
 </auro-accordion>
 
+## Header Slot Example
```
Comment (blackfalcon) [inline demo/demo.md:71]: This example needs to be moved down near the footer slot example.

File: demo/demo.md [hunk 2]
```diff
@@ -68,6 +68,46 @@ This example illustrates a mainline nonstop flight from SEA to EWR (SEAEWR)
...
+
+This example illustrates a mainline nonstop flight from SEA to EWR (SEAEWR)
```
Comment (blackfalcon) [inline demo/demo.md:73]: The example is showing more than a flight from SEA to EWR. There is no information to point the reader to the fact that the additional slots are being used and why.

File: demo/demo.md [hunk 3]
```diff
@@ -68,6 +68,46 @@ This example illustrates a mainline nonstop flight from SEA to EWR (SEAEWR)
...
+  <auro-flight
+    flights='["AS 8"]'
+    duration="5h 25m"
+    daysChanged="0"
+    departureTime="7:30 am"
+    departureStation="SEA"
+    arrivalTime="3:55 pm"
+    arrivalStation="EWR"
+    >
+    <auro-flightline></auro-flightline>
```
Comment (blackfalcon) [inline demo/demo.md:105]: The demo code does not match the used code.

File: docs/api.md [hunk 4]
```diff
@@ -53,7 +53,9 @@ Please DO NOT modify unit tests pertaining to DoT regulations without contacting
...
 
-| Name      | Description                                      |
-|-----------|--------------------------------------------------|
-| `default` | anticipates `<auro-flightline>` instance to fill out the flight timeline |
-| `footer`  | Lower section allowing for tertiary content to be attributed to the element. Per **DoT Regulations** do NOT edit the styles contained within this slot. |
+| Name          | Description                                      |
+|---------------|--------------------------------------------------|
+| `default`     | anticipates `<auro-flightline>` instance to fill out the flight timeline |
+| `footer`      | Lower section allowing for tertiary content to be attributed to the element. Per **DoT Regulations** do NOT edit the styles contained within this slot. |
+| `leftHeader`  | Header on top of the departure station's time    |
```
Comment (blackfalcon) [inline docs/api.md:60]: I don't feel that these API names are accurate. This not a generic 'header' element for anything to be placed in. 

What is the use case for the additional content?

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: src/style-flight.scss [hunk 5]
```diff
@@ -20,3 +20,8 @@
 slot[name=footer] {
   color: var(--auro-color-text-secondary-on-light);
 }
+
+.header-container {
```
Comment (blackfalcon) [inline src/style-flight.scss:24]:
```suggestion
.headerContainer {
```

`.header-container` implies that `container` is a child of `header`. 

https://auro.alaskaair.com/webcorestylesheets/conventions

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Fix this selector and you are golden.

File: src/style-flight.scss (hunk 5 above)
Comment (gusnaughton) [reply src/style-flight.scss:24]: #fixed

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**